vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/TextField.java

```diff
@@ -450,12 +452,6 @@ public String getValue() {
         return super.getValue();
     }
 
-    @Override
-    protected void setModelValue(String newModelValue, boolean fromClient) {
-        super.setModelValue(newModelValue, fromClient);
-        setInvalid(getValidationSupport().isInvalid(newModelValue));
```

Comment:
The problem with this approach was, that the `super.setModelValue` triggers all the `ValueChangeListener`s. One of those is added by Binder to run the validation. The component's own server-side validation runs after that, on the second line.

Another problem was, that this `setModelValue` can execute also when the old value is equal to the new value.